pkg/agent/config/config.go

```diff
@@ -365,3 +373,29 @@ func HostnameCheck(cfg cmds.Agent) error {
 	}
 	return fmt.Errorf("Timed out waiting for hostname %s to be resolvable: %v", hostname, err)
 }
+
+func getIPFromInterface(flannelIfaceName string) (string, error) {
+	flannelIface, err := sysnet.InterfaceByName(flannelIfaceName)
+	if err != nil {
+		return "", err
+	}
+	addrs, err := flannelIface.Addrs()
+	if err != nil {
+		return "", err
+	}
+	if flannelIface.Flags&sysnet.FlagUp == 0 {
+		return "", fmt.Errorf("the interface %s is not up", flannelIfaceName)
+	}
+	for _, addr := range addrs {
```

Comment:
Can we change this so that if we find more than 1 ipv4 address we error out?  So we force the user to only have one ipv4 on the iface and we use that.  This is similar logic to how flannel already works https://github.com/ibuildthecloud/flannel/blob/823afe66b2266bf71f5bec24e6e28b26d70cfc7c/pkg/ip/iface.go#L157